sdk/core/azure-core/src/main/java/com/azure/core/util/FluxUtil.java

```diff
@@ -83,6 +84,74 @@ private static void accept(ByteArrayOutputStream byteOutputStream, ByteBuffer by
         return byteArray;
     }
 
+    /**
+     * Converts an {@link InputStream} into a {@link Flux} of {@link ByteBuffer} using a chunk size of 4096.
+     * <p>
+     * Given that {@link InputStream} is not guaranteed to be replayable the returned {@link Flux} should be considered
+     * non-replayable as well.
+     * <p>
+     * If the passed {@link InputStream} is {@code null} {@link Flux#empty()} will be returned.
+     *
+     * @param inputStream The {@link InputStream} to convert into a {@link Flux}.
+     * @return A {@link Flux} of {@link ByteBuffer ByteBuffers} that contains the contents of the stream.
+     */
+    public static Flux<ByteBuffer> toFluxByteBuffer(InputStream inputStream) {
+        return toFluxByteBuffer(inputStream, 4096);
+    }
+
+    /**
+     * Converts an {@link InputStream} into a {@link Flux} of {@link ByteBuffer}.
+     * <p>
+     * Given that {@link InputStream} is not guaranteed to be replayable the returned {@link Flux} should be considered
+     * non-replayable as well.
+     * <p>
+     * If the passed {@link InputStream} is {@code null} {@link Flux#empty()} will be returned.
+     *
+     * @param inputStream The {@link InputStream} to convert into a {@link Flux}.
+     * @param chunkSize The requested size for each {@link ByteBuffer}.
+     * @return A {@link Flux} of {@link ByteBuffer ByteBuffers} that contains the contents of the stream.
+     * @throws IllegalArgumentException If {@code chunkSize} is less than or equal to {@code 0}.
+     */
+    public static Flux<ByteBuffer> toFluxByteBuffer(InputStream inputStream, int chunkSize) {
+        if (chunkSize <= 0) {
+            throw new IllegalArgumentException("'chunkSize' must be greater than 0.");
+        }
```

Comment:
A method returning a reactive publisher shouldn't throw exceptions. It makes chaining of reactive operators difficult as the caller will have to handle errors in 2 ways. Return Flux.error() instead.

Reply:
Changed to `Flux.error`